lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -750,8 +764,21 @@ void TypeLoweringVisitor::lowerSAWritePath(Operation *op,
   }
 }
 
+static bool
+canLowerConnect(FConnectLike op,
+                PreserveAggregate::PreserveMode aggregatePreservationMode) {
+  auto destType = op.getDest().getType();
+  if (!destType.isa<RefType>())
+    return true;
+
+  return !(destType.isa<RefType>() &&
+           isPreservableAggregateType(destType, aggregatePreservationMode));
```

Comment:
I think the latter check of `!destType.isa<RefType>() || !isPreservableAggregateType` can just be `return `!isPreservableAggregateType` as `!destType.isa<RefType>()` is always false given the exit above. You may want to check my De Morgan-ing.